pkg/registry/core/adapters/nse_registry.go

```diff
@@ -44,6 +44,9 @@ func (n *networkServiceEndpointRegistryServer) Find(query *registry.NetworkServi
 	}
 
 	for {
+		if client == nil {
```

Comment:
please move this check outside of the loop.

Reply:
ok, fixed